extensions/applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -9,149 +9,170 @@ import {
     IPlugin,
     IAppInsightsCore, 
     CoreUtils,
-    IDiagnosticLogger,
     LoggingSeverity,
     _InternalMessageId,
-    DiagnosticLogger
+    BaseTelemetryPlugin,
+    IProcessTelemetryContext
 } from '@microsoft/applicationinsights-core-js';
 import { ConfigurationManager, IDevice, IExceptionTelemetry, IAppInsights, SeverityLevel, AnalyticsPluginIdentifier  } from '@microsoft/applicationinsights-common';
 import DeviceInfo from 'react-native-device-info';
 
 import { INativeDevice, IReactNativePluginConfig } from './Interfaces';
+import dynamicProto from '@microsoft/dynamicproto-js';
 
-export class ReactNativePlugin implements ITelemetryPlugin {
+export class ReactNativePlugin extends BaseTelemetryPlugin {
 
     identifier: string = 'AppInsightsReactNativePlugin';
     priority: number = 140;
     _nextPlugin?: ITelemetryPlugin;
-    private _initialized: boolean = false;
-    private _device: INativeDevice;
-    private _config: IReactNativePluginConfig;
-    private _analyticsPlugin: IAppInsights;
-    private _logger: IDiagnosticLogger;
-    private _defaultHandler;
 
     constructor(config?: IReactNativePluginConfig) {
-        this._config = config || this._getDefaultConfig();
-        this._device = {};
-        this._logger = new DiagnosticLogger();
-    }
-
-    public initialize(
-        config?: IReactNativePluginConfig | object, // need `| object` to coerce to interface
-        core?: IAppInsightsCore,
-        extensions?: IPlugin[]
-    ) {
-        if (!this._initialized) {
-            const inConfig = config || {};
-            const defaultConfig = this._getDefaultConfig();
-            for (const option in defaultConfig) {
-                this._config[option] = ConfigurationManager.getConfig(
-                    inConfig as any,
-                    option,
-                    this.identifier,
-                    this._config[option]
-                );
+        super();
+
+        let _device: INativeDevice = {};
+        let _config: IReactNativePluginConfig = config || _getDefaultConfig();
+        let _analyticsPlugin: IAppInsights;
+        let _defaultHandler;
+    
+        dynamicProto(ReactNativePlugin, this, (_self, _base) => {
```

Comment:
Also, as I was touching the class -- implemented with dynamicProto() for minimifaction.

NOTE: The private properties are now within the constructor's closure -- so they will NOT be available on the instance for any form of overloading... If anyone is aware of these properties being used directly (even though they are private) then we will need to make them protected / public to properly support that usage pattern.